gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -157,13 +158,37 @@ def _read_variants(pipeline, known_args):
   return variants
 
 
-def _use_beam_header_merger(known_args):
+def _beam_merge_headers(known_args, pipeline_args):
   if known_args.representative_header_file:
-    return False
+    return
+
   match_results = FileSystems.match([known_args.input_pattern])
   if not match_results:
-    return False
-  return len(match_results[0].metadata_list) > _USE_HEADER_MERGER_THRESHOLD
+    return None
+
+  paths = [f.path for f in match_results[0].metadata_list]
+  if len(paths) > _USE_BEAM_READ_HEADER_THRESHOLD:
```

Comment:
Please add a logic to automatically set "pipeline_args.optimize_for_large_inputs = True" if the len is > _USE_BEAM_READ_ALL_HEADER_THRESHOLD

Reply:
Done.